src/DIRAC/Resources/Computing/HTCondorCEComputingElement.py

```diff
@@ -367,42 +430,82 @@ def getJobStatus(self, jobIDList):
             condorIDs[job] = jobID
 
+        self.tokenFile = None
+
         qList = []
         for _condorIDs in breakListIntoChunks(condorIDs.values(), 100):
 
             # This will return a list of 1245.75 3
-            status, stdout_q = subprocess.getstatusoutput(
-                "condor_q {} {} -af:j JobStatus ".format(self.remoteScheddOptions, " ".join(_condorIDs))
-            )
+            cmd = ["condor_q"]
+            cmd.extend(self.remoteScheddOptions.strip().split(" "))
+            cmd.extend(_condorIDs)
+            cmd.extend(["-af:j", "JobStatus"])
+            result = self._executeCondorCommand(cmd, keepTokenFile=True)
+            if not result["OK"]:
+                if self.tokenFile:
+                    os.remove(self.tokenFile)
+                    self.tokenFile = None
+                return S_ERROR("condor_q failed completely: %s" % result["Message"])
+            status, stdout, stderr = result["Value"]
             if status != 0:
-                return S_ERROR(stdout_q)
-            _qList = stdout_q.strip().split("\n")
+                if self.tokenFile:
+                    os.remove(self.tokenFile)
+                    self.tokenFile = None
+                return S_ERROR(stdout + stderr)
+            _qList = stdout.strip().split("\n")
             qList.extend(_qList)
 
             # FIXME: condor_history does only support j for autoformat from 8.5.3,
             # format adds whitespace for each field This will return a list of 1245 75 3
-            # needs to cocatenate the first two with a dot
+            # needs to concatenate the first two with a dot
             condorHistCall = "condor_history {} {} -af ClusterId ProcId JobStatus".format(
                 self.remoteScheddOptions,
                 " ".join(_condorIDs),
-            )
+            ).split()
 
-            treatCondorHistory(condorHistCall, qList)
+            self._treatCondorHistory(condorHistCall, qList)
 
         for job, jobID in condorIDs.items():
 
             pilotStatus = parseCondorStatus(qList, jobID)
             if pilotStatus == "HELD":
                 # make sure the pilot stays dead and gets taken out of the condor_q
-                _rmStat, _rmOut = subprocess.getstatusoutput(f"condor_rm {self.remoteScheddOptions} {jobID} ")
-                # self.log.debug( "condor job killed: job %s, stat %s, message %s " % ( jobID, rmStat, rmOut ) )
+                cmd = f"condor_rm {self.remoteScheddOptions} {jobID}".split()
+                _result = self._executeCondorCommand(cmd, keepTokenFile=True)
                 pilotStatus = PilotStatus.ABORTED
 
             resultDict[job] = pilotStatus
 
+        if self.tokenFile:
+            os.remove(self.tokenFile)
+            self.tokenFile = None
+
         self.log.verbose("Pilot Statuses: %s " % resultDict)
         return S_OK(resultDict)
 
+    def _treatCondorHistory(self, condorHistCall, qList):
```

Comment:
Did you move this from `DIRAC.Resources.Computing.BatchSystems.Condor.treatCondorHistory`? Why?
Anyway, you can remove the import now but mostly remove the other code to avoid duplications?